### DISCUSSION TITLE
How can I apply a cube rotation so the internal state of the cube matches my preferred orientation?

### DISCUSSION THREAD

**Discussion by FoxMcWeezer** [cs0x7f/cstimer#343] [Q&A] [answered]
For example, I'd like to perform an equivalent statement to

```
var cc = new mathlib.CubieCube();
cc.selfMoveStr("y");
...
cc.toFaceCube();
```

such that ```cc.toFaceCube();``` outputs the cube definition string in my preferred orientation (cube rotation on the y axis. Thanks in advance.

@cs0x7f

**Comment by cs0x7f** [comment, marked as answer]
In fact, its internal state orientation has been changed (determined by cc.ori, the default value is 0).
If other moves are continued, the state transformation will be different from the original orientation.
Only when toFaceCube is output, we will turn the center back.

If you just want toFaceCube to output the cube in a new orientation, you need to make some changes in CubieCube.prototype.toFaceCube.
